### ISSUE TITLE
[Feature request] Search season packs when cutoff unmet [not planned on decluttarr ->sonarr] 

### ISSUE THREAD

**Issue by adrianipopescu** [ManiMatter/decluttarr#212]
Any chance I can have the tool search for a whole season in sonarr instead of individual searches per episode?

**Comment by ManiMatter** [comment]
Good idea. Im limited on time, would you like to contribute with a PR?

**Comment by adrianipopescu** [comment]
let me try to understand the codebase and will give it a shot

just be gentle during review

**Comment by ManiMatter** [comment]
Sonarr confirmed that season packs are only grabbed if the entire season pack would be imported.
Meaning: If you have at least one file that is off the same or better quality than the season pack, the season pack wouldn't be grabbed / imported. 
If you search for an upgrade of a file and it finds a season pack that is an upgrade to all, then it would be grabbed.

Therefore, what you ask for is already there, natively built into sonarr.
What isn't there is that partial upgrades would be fetched.

For this to work, a change in sonarr would be required; even if decluttarr sent a search, and even if it found a season pack that contains 9/10 files that you don't have, or in a better quality, it would still not grab them because you do have the 1 file that is of the same or better quality..

Thus nothing to be done here on decluttarr, closing this. 
Note that there's a already a feature request open with sonarr on this, you might want to follow it there:
https://github.com/Sonarr/Sonarr/issues/6378
